v2/api_error.go

```diff
@@ -168,6 +168,33 @@ func (a *APIError) GRPCStatus() *status.Status {
 	return a.status
 }
 
+// Reason returns the reason in an ErrorInfo.
+// If ErrorInfo is empty, it returns an empty string.
+func (a *APIError) Reason() string {
+	if a.details.ErrorInfo.GetReason() != "" {
```

Comment:
Check if `a.details.ErrorInfo != nil` instead to avoid a `panic`.

Checking if it is empty doesn't do anything different than just returning `""` - if you returned `Reason` and it was empty, it would be `""`.